Riot API V3 + multicurl management

Canisback opened the pull request:
I updated for the Riot API V3.
I deleted all soon to be deprecated function and added the new ones.

I added a multi curl function to handle multiple requested at the same time. I use it for example in the getMatch function, as the timeline can only be obtained by a separated call now, so if timeline is needed, both calls (match and timeline) are done at the same time. Usefull too when calling a bunch of matches.

Now that the API supports it, I moved the api key to the header, as it is easier to use with multiple arguments queries, also for cache, you can get a shared cache with other applications with different keys without redundant call only because of different key in url.

kevinohashi commented:
Thanks for this awesome contribution. I haven't had a chance to test it out yet, hopefully next week I can. If anyone can tell me it works, I'll happily merge it sooner.

Talha-T commented:
Yeah I started using this, and works perfectly; but getFreeChampions is changed to getFreeChampion

Canisback commented:
I fixed the getFreeChampions function to the right name and added some tests.
By the way, I just bumped into this problem today, the multi curl seems to have a lot of trouble when you call  more than ~150 queries at the same time.

kevinohashi commented:
Thanks for your contribution @Canisback!
